src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -50,10 +53,26 @@ public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount co
                 .getAttribute(AttributeName.TRANSACTION.getValue());
 
         String requestId = request.getHeader("X-Request-Id");
-        ResponseObject result = companyAccountService
-                .createCompanyAccount(companyAccount, transaction, requestId);
-        return apiResponseMapper
-                .map(result.getStatus(), result.getData(), result.getErrorData());
+
+        ResponseEntity responseEntity;
+        try {
+            CompanyAccount companyAccountInserted = companyAccountService
+                    .createCompanyAccount(companyAccount, transaction, requestId);
+            responseEntity = apiResponseMapper
+                    .map(ResponseStatus.SUCCESS_CREATED, companyAccountInserted);
+        } catch (ApiErrorResponseException aere) {
+            LOGGER.error(aere);
+            responseEntity = apiResponseMapper.map(ResponseStatus.TRANSACTION_PATCH_ERROR);
+        } catch (DataException se) {
+            LOGGER.error(se);
+            if (se.getCause() instanceof DuplicateKeyException) {
+                responseEntity = apiResponseMapper.map(ResponseStatus.DUPLICATE_KEY_ERROR);
```

Comment:
same comment here - response status should not be used for mapping error responses